src/wp-includes/class-wp-image-editor.php

```diff
@@ -471,7 +471,10 @@ public function generate_filename( $suffix = null, $dest_path = null, $extension
 			$suffix = "-{$suffix}";
 		}
 
-		// When the mime type being generated is different from the source, add the extension to the suffix to ensure uniqueness.
+		// When the file extension being generated doesn't match the image file extension,
+		// add the extension to the suffix to ensure a unique file name. Prevents
+		// name conflicts when a single image type can have multiple extensions,
+		// eg. .jpg, .jpg and .jpeg are all valid JPEG extensions.
```

Comment:
@adamsilverstein You have `.jpg` in here twice, one needs to be `.jpe` :)

Reply:
Fixed in https://github.com/WordPress/wordpress-develop/pull/2393/commits/5dbcd64816e39b4bd27fc625a8c3dd21823285db